api/resourcedeleter.go

```diff
@@ -0,0 +1,123 @@
+package api
+
+import (
+	"k8s.io/client-go/kubernetes"
+	k8smeta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"fmt"
+	redisclient "github.com/spotahome/redis-operator/client/k8s/clientset/versioned/typed/redisfailover/v1alpha2"
+	"k8s.io/client-go/rest"
+)
+
+
+func deleteK8sResouces(namespace string, deployName string, k8sClient kubernetes.Interface) error {
+	deploymentDeleteOption := k8smeta.DeletePropagationForeground
+
+	err := k8sClient.CoreV1().Services(namespace).Delete(deployName, &k8smeta.DeleteOptions{})
```

Comment:
Whenever we have a function like this that returns a single error value, it might be suitable with the if err := func(); err != nil { //body } idiom. Here and in the following delete operations this might be suitable. Also where this function we are in are called, we can do the same.

Reply:
Jeg lurer litt på fordelen med den stilen?

Reply:
by using this idiom, you avoid the explicit function scope assignment of err in fewer lines of code

```
if err := func(); err != nil {
    //body
}
```

vs

```
err := func()
if err != nil {
    //body
}
```

Reply:
Hum...hadde det vært et kort funksjonsnavn, så hadde jeg sett fordelen, men syns kanskje `err != nil` blir litt gjemt nå.

```
err := k8sClient.CoreV1().Services(namespace).Delete(deployName, &k8smeta.DeleteOptions{}); err != nil {
    // body
}

Reply:
That doesnt compile, you need the if. A long or short function call statement doesn't matter imo. 

Both are valid, but this makes more sense to me. Also [recommended in Effective Go](https://golang.org/doc/effective_go.html#if)

Reply:
not a big issue here, but this would also allow for the resource to be garbage collected/popped from stack as it is no longer within the scope after the `if` statement, and doesn't pollute the scope of the `func`